Feature/merge server client webpack

jbhatab opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/shakacode/react_on_rails/398)
<!-- Reviewable:end -->

justin808 commented:
Some comments.

---

Reviewed 39 of 39 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[lib/generators/USAGE, line 3 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5TpWXh301vsQe_lna:-KG5TpWXh301vsQe_lnb:498532620) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/USAGE#L3)):*
trailing spaces

---

*[lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt, line 2 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5UYTncpy_eIoT5Rt4:-KG5UYTose-jlLNlYSZp:417313250) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt#L2)):*
Add named param prerender: false

---

*[lib/generators/react_on_rails/templates/base/base/client/webpack.config.js, line 38 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5UwX48h8rX-0MpVzN:-KG5UwX48h8rX-0MpVzO:-367373384) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/client/webpack.config.js#L38)):*
We're not using jquery by default

---

*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VDYu-YOGlOtXNepX:-KG5VDYu-YOGlOtXNepY:325338467) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L14)):*
This makes some parts of the code expect that server rendering is on. This is definitely an issue.

This needs to be "" if not server rendering.

search code for server_bundle_js_file and you'll see

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 11 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5WPsf8dsEPjMA9k0C:-KG5WPsf8dsEPjMA9k0D:2136018752) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L11)):*
the function would not use props


also need to always pass the `railsContext` as the second param for components and stores

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KG5W88mqgpk5LwIkmoo:-178244202) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
worth considering if we want this to be:

const store = ReactOnRails.getStore("myStore");

that way you can have multiple react components use the same store

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 22 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VtBHiz-8IO09VGWS:-KG5VtBIHf9oVeT97g9o:-1389175974) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L22)):*
line endings?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-:-KG5YwcWRwIF-KJHALjU:-1683452685)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt, line 2 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5UYTncpy_eIoT5Rt4:-KG5j6HNFoEFUaDKwCRM:677588487) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt#L2)):*
done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/client/webpack.config.js, line 38 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5UwX48h8rX-0MpVzN:-KG6PhCr3Fc43f6dP5Ia:-285194829) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/client/webpack.config.js#L38)):*
removing it completely. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VDYu-YOGlOtXNepX:-KG6SQPm9ftKCrcRJN2N:-87711433) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L14)):*
Looks like actually a few lines of code on the backend will need to change that are tied to the words server/client.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KG6gDlso2zoKaUAFcvD:-30812356) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
yeah agreed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VDYu-YOGlOtXNepX:-KG6gKUeKGpsSnEWw3EP:717796638) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L14)):*
@justin808 I dove more into the lib code and it seems it may be more complicated than we hoped. Right now the code is tied to the name's server for a bundle name and there are lines that are strictly tied to npm run build:server and npm run build:client.

```
response = `pgrep -fl 'bin/webpack\s(\\-w|\\-\\-watch)\s\\-\\-config\s.*#{type}.*\\.js'`
```

and 

```
build_output = `cd client && npm run build:#{type}`
```

Now I think we have a few options. 

1. Always require that the server bundle is also used in the client. Now there could be multiple generated bundles, but one of the client bundles would have to be used for the server as well. 

2. Not sure if this will work but having npm run build:server and npm run build:client but they just pass variables to the same config to spit out different files.  At that point it may make more sense to just go with 2 configs though.

3. we add a dynamic element that allows people to add on npm commands to do whatever like server render a separate config if needed.

I'm personally in favor of just going for a single config since most people will just be putting the entire bundle in their header which is awesome with turbolinks. And if you need to optmize, then you can still do vendor chunks as needed which would still play nice with all of this. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/USAGE, line 3 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5TpWXh301vsQe_lna:-KG6gAaBS5Pyoai9MqCA:-499146009) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/USAGE#L3)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
Just changed the webpack compilation checks for the tests. Now it just checks to see if any stale files exist and then runs a new config variable we added if they do. we default to 'npm run build'.

justin808 commented:
Reviewed 1 of 39 files at r1, 9 of 12 files at r2, 6 of 11 files at r3.
Review status: 40 of 48 files reviewed at latest revision, 8 unresolved discussions.

---

*[lib/react_on_rails.rb, line 17 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk4MT2ZU7keWlbhzBP:-KGk4MT2ZU7keWlbhzBQ:-777788716) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails.rb#L17)):*
Where did this go?

---

*[lib/generators/react_on_rails/templates/base/base/package.json.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk4Zj7KR4VKU7HctiL:-KGk4Zj7KR4VKU7HctiM:-239557509) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/package.json.tt#L12)):*
why remove these?

---

*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VDYu-YOGlOtXNepX:-KGk58km1q6ZD8M02_Qn:1628195102) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L14)):*
what if we just check for a "webpack -w" ?

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 11 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5WPsf8dsEPjMA9k0C:-KGk5TJd4Fk9abSlEQ1H:453173392) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L11)):*
Are you not finished with this one?

We should pass `_props` and `_railsContext` and use neither, as they will be used for store initialization.

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KGk5cPAt8N2G7G3vYkb:-833311332) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
Not done yet

---

*[lib/react_on_rails/test_helper.rb, line 57 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk5onlG2igoTj1EL1R:-KGk5onmJ6zzqz0TR2Ds:502826998) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails/test_helper.rb#L57)):*
Why are we removing this? Let's keep it.

---

*[lib/react_on_rails/test_helper/ensure_assets_compiled.rb, line 49 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk5wJaG7gtvjOy6H4A:-KGk5wJaG7gtvjOy6H4B:106673143) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails/test_helper/ensure_assets_compiled.rb#L49)):*
Let's not remove all of this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Got some questions. Also Please see https://github.com/shakacode/react_on_rails/blob/master/docs%2Fcontributor-info%2Fcontributing.md. Notably, we need a changelog entry.

---

Review status: 40 of 48 files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-:-KGnI_THOdqyMbZoWAx4:472317399)*
<!-- Sent from Reviewable.io -->

justin808 commented:
@jbhatab Is this almost ready?

robwise commented:
Reviewed 1 of 39 files at r1, 4 of 12 files at r2, 7 of 11 files at r3.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[lib/react_on_rails.rb, line 17 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk4MT2ZU7keWlbhzBP:-KH7mOwz_MEAHRXFtoFs:1044417279) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails.rb#L17)):*
We deleted it, it's super super brittle/unmaintainable and was not working anyway. Simply checking if the assets are up to date or not should be sufficient. There is only the unlikely case where a user changes something inside of the client code and is able to start the tests so fast that they kick off the ensure assets compiled process before the webpack process has finished compiling, in which case it just re-compiles so not that painful.

---

*[lib/generators/react_on_rails/templates/base/base/package.json.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk4Zj7KR4VKU7HctiL:-KH7n1-PVOHasuXmLCkZ:-95343918) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/package.json.tt#L12)):*
it isn't DRY and therefore is extremely brittle

---

*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VDYu-YOGlOtXNepX:-KH7nVDgTH4A524xptlo:-679449585) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L14)):*
we got rid of the process checker, this is a moot point now, right @jbhatab ?

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 11 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5WPsf8dsEPjMA9k0C:-KH7pNBwWTznFuZql1aA:-323441154) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L11)):*
see below comment

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KH7pMEfrUC9usM_Fvzh:-618725379) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
using the shared store is a very advanced, niche feature. I don't think we should generate it in a simple hello world example, as it's just going to confuse people. We should keep it as vanilla as possible and leave the advanced redux store feature for the API docs and possibly a recipe section of the docs.

---

*[lib/react_on_rails/test_helper.rb, line 57 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk5onlG2igoTj1EL1R:-KH7o5zbhGK6-ujqP7Q9:-1742223608) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails/test_helper.rb#L57)):*
See previous comment

---

*[lib/react_on_rails/test_helper/ensure_assets_compiled.rb, line 49 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk5wJaG7gtvjOy6H4A:-KH7oLQaEioQguqOgQD_:1228075795) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails/test_helper/ensure_assets_compiled.rb#L49)):*
this code is extremely brittle and not surprisingly was broken. Now that it's fixed, I highly suggest keeping it this way. It's much easier to read and works correctly now

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Reviewed 26 of 39 files at r1, 10 of 12 files at r2, 8 of 11 files at r3.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt, line 2 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5UYTncpy_eIoT5Rt4:-KH8-xKzTbiU1bjg6pLl:-985799675) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt#L2)):*
why do we want to explicitly override the default the user sets in the config here? Isn't that a little assuming? It's also one additional thing they have to understand when first getting used to our API

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/react_on_rails.rb, line 17 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk4MT2ZU7keWlbhzBP:-KHBd6HD2aHiXZGeF1gZ:-1996135412) ([raw file](https://github.com/shakacode/react_on_rails/blob/999561952be5c1ab84a608b1f3d84e619dac86c2/lib/react_on_rails.rb#L17)):*
Yeah, that was the general logic. It was too fragile and the scenarios where it even mattered we're very rare. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/package.json.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KGk4Zj7KR4VKU7HctiL:-KHBdIxF8Qeb4Holj5ds:1003304187) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/package.json.tt#L12)):*
Even if we do keep these, we would condense them down to build and build:dev vs having all of them, but I think we should keep all of the npm commands that are related to the client directory in the package.json client directory. This adds a layer of magic that may confuse new people into thinking it's necessary to use react_on_rails.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt, line 2 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5UYTncpy_eIoT5Rt4:-KHBddMC-AgDVQ494NNB:-797632597) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/app/views/hello_world/index.html.erb.tt#L2)):*
We can change this later if people complain. I think it's pretty minor.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 14 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VDYu-YOGlOtXNepX:-KHBdriK4bnA9An2mliX:-377800951) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L14)):*
Yup, we removed the process checker. It was very fragile and already brought up a bug where the core functionality of checking to see if the webpack file was stale broke. Also if people change their npm commands then it becomes useless. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KHBeCt9iMeIEvcnFkke:2039372779) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
@justin808 I want to add this in a separate PR or in the component generator PR. This PR is already getting a little big and we should merge this in first. I agree that it's a cool feature to show off. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 22 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5VtBHiz-8IO09VGWS:-KHBebQT5sCfPdmTpT9P:-1146861887) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L22)):*
show's a line ending for me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Reviewed 1 of 16 files at r6.
Review status: 37 of 61 files reviewed at latest revision, 11 unresolved discussions.

---

*[CHANGELOG.md, line 13 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIum-NIRnJXqVFRuQx:-KHIum-NIRnJXqVFRuQy:1164787168) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/CHANGELOG.md#L13)):*
> ```Markdown
> ##### Removed
> - Server rendering is no longer an option in the generator and is always accessible [#398](https://github.com/shakacode/react_on_rails/pull/398).
> - removed lodash, jquery, and loggerMiddleware from the generatred code [#398](https://github.com/shakacode/react_on_rails/pull/398).
> ```

`generatred` typo

---

*[CHANGELOG.md, line 14 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIujp_WSuKp5_0At6a:-KHIujpapqToVJDALgd6:-643679756) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/CHANGELOG.md#L14)):*
> ```Markdown
> - Server rendering is no longer an option in the generator and is always accessible [#398](https://github.com/shakacode/react_on_rails/pull/398).
> - removed lodash, jquery, and loggerMiddleware from the generatred code [#398](https://github.com/shakacode/react_on_rails/pull/398).
> - removed webpack watch check for test helper automatic compiltion [#398](https://github.com/shakacode/react_on_rails/pull/398).
> ```

`compilition` typo

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Review status: 37 of 61 files reviewed at latest revision, 13 unresolved discussions.

---

*[lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt, line 3 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvLsaZalOpWA2Jf_g:-KHIvLsbBsmDiDcVMiyx:1104451061) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt#L3)):*
> ```
> web: rails s
> client: sh -c 'rm app/assets/webpack/* || true && cd client && npm run build:dev'
> node: sh -c 'cd client/node && npm start'
> ```

this one seems to have an eof line return problem

---

*[lib/generators/react_on_rails/templates/base/base/client/.DS_Store, line 0 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvRKdZoXfnl5CesMq:-KHIvRKdZoXfnl5CesMr:-1020919916) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/.DS_Store#L0)):*
we need to add `.DS_Store` to gitignore and delete from this commit

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 1 of 16 files at r6.
Review status: 38 of 61 files reviewed at latest revision, 15 unresolved discussions.

---

*[README.md, line 266 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIwA0jA11Fho2a3C1G:-KHIwA0kyXDfaX4JBca0:-1865903358) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/README.md#L266)):*
> ```Markdown
> Place your JavaScript code inside of the provided `client/app` folder. Use modules just as you would when using webpack alone. The difference here is that instead of mounting React components directly to an element using `React.render`, you **expose your components globally and then mount them with helpers inside of your Rails views**.
> 
> This is an example of how to expose your component. Can be seen in the `HelloWorldApp.jsx` file.
> ```

grammar

---

*[README.md, line 272 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIwFxd-2CiHc3xyP8O:-KHIwFxd-2CiHc3xyP8P:2041231880) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/README.md#L272)):*
> ```Markdown
> import ReactOnRails from 'react-on-rails';
> ReactOnRails.register({ HelloWorld });
> ```
> ```

we need to note that many apps may require separate server initialization files.

I don't appreciate having this stuff I wrote totally being deleted:

```
  In general, you may want different initialization for your server rendered components.
 
See below section on how to setup redux stores that allow multiple components to talk to the same store.
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt, line 3 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvLsaZalOpWA2Jf_g:-KHIwZnIeH6gJKcTcHku:-1620474789) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt#L3)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> this one seems to have an eof line return problem</details>
fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/client/.DS_Store, line  \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvRKdZoXfnl5CesMq:-KHIxB2byk8Cku7eXZsO:-1620474753) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/.DS_Store#L)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> we need to add `.DS_Store` to gitignore and delete from this commit</details>
fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 3 of 7 files at r4, 3 of 8 files at r5, 6 of 16 files at r6.
Review status: 50 of 61 files reviewed at latest revision, 17 unresolved discussions.

---

*[docs/basics/generator.md, line 17 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIwkmfvNXuLm-QBrbz:-KHIwkmgtIlDkYaIg57q:-607475697) ([raw file](https://github.com/shakacode/react_on_rails/blob/949117978e75babd50f5f5d147f49c50674cf9cd/docs/basics/generator.md#L17)):*
> ```Markdown
> Options:
>   -R, [--redux], [--no-redux]                          # Install Redux gems and Redux version of Hello World Example
>   -S, [--server-rendering], [--no-server-rendering]    # Add necessary files and configurations for server-side rendering
> ```

Do we document how to turn on server rendering? The code generated should have some comments on this.

---

*[lib/generators/react_on_rails/base_generator.rb, line 97 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIx8tiu0MjGO0ZxPw7:-KHIx8tiu0MjGO0ZxPw8:498090155) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/base_generator.rb#L97)):*
> ```Ruby
>       def template_assets_rake_file
>         template("base/base/lib/tasks/assets.rake.tt", "lib/tasks/assets.rake")
>         copy_file("base/base/lib/tasks/load_test.rake", "lib/tasks/load_test.rake")
> ```

Why is the load test part of the generators, @jbhatab @robwise? 

---

*[lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt, line 3 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvLsaZalOpWA2Jf_g:-KHIxGC7cl_8WkOKCeOD:-915113734) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt#L3)):*
<details><summary><i>Previously, jbhatab (Blaine Hatab) wrote…</i></summary>
> fixed
> </details>
still there on reviewable

---

*[lib/generators/react_on_rails/templates/base/base/client/.DS_Store, line  \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvRKdZoXfnl5CesMq:-KHIxJNTKhJ89HDLjlNZ:-6107870) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/.DS_Store#L)):*
<details><summary><i>Previously, jbhatab (Blaine Hatab) wrote…</i></summary>
> fixed.</details>
Rob, We don't add .DS_STORE to .gitignore 

that should be in every mac developer's global gitignore.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Review status: 50 of 61 files reviewed at latest revision, 18 unresolved discussions.

---

*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 12 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIxU7PcwGluqhU6AYx:-KHIxU7PcwGluqhU6AYy:-1859188630) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L12)):*
> ```
>     "build:test": "webpack --config webpack.config.js",
>     "build:production": "NODE_ENV=production webpack --config webpack.config.js",
>     "build:dev": "webpack -w --config webpack.config.js"
> ```

test, prod, dev

or test, production, development

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 1 of 7 files at r4, 4 of 8 files at r5, 8 of 16 files at r6.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[docs/basics/generator.md, line 17 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIwkmfvNXuLm-QBrbz:-KHIxc2WEtURZ0HgOR3o:1748965079) ([raw file](https://github.com/shakacode/react_on_rails/blob/949117978e75babd50f5f5d147f49c50674cf9cd/docs/basics/generator.md#L17)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> Do we document how to turn on server rendering? The code generated should have some comments on this.</details>
Yes, we have it in the component API, `prerender: true` turns on server rendering for that component. That's all there is to it now!

---

*[lib/generators/react_on_rails/base_generator.rb, line 97 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIx8tiu0MjGO0ZxPw7:-KHIy48_RxH6eM3jAexd:-2011945671) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/base_generator.rb#L97)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> Why is the load test part of the generators, @jbhatab @robwise? </details>
I have no idea what either of these changes these are

---

*[lib/generators/react_on_rails/templates/base/base/client/.DS_Store, line  \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvRKdZoXfnl5CesMq:-KHIyCB2UP3lW41CcEPL:401056544) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/.DS_Store#L)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> Rob, We don't add .DS_STORE to .gitignore 
> 
> that should be in every mac developer's global gitignore.</details>
it _should_ be, but as evidenced by the fact that we are having this conversation, it obviously isn't, and therefore it would be prudent to prevent this from ever happening again by adding it to the gitignore. Considering the cost is nothing, and there is a benefit, it's a no-brainer. But this is sort of tangential to the point of the comment, which was "let's not commit a DS_Store file"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 12 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIxU7PcwGluqhU6AYx:-KHIyqRq2xzpNaQ6Ldbc:1991939039) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L12)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> test, prod, dev
> 
> or test, production, development
> 
> </details>
In that case I vote for the longer words so it will be just like the Rails envs

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[README.md, line 272 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIwFxd-2CiHc3xyP8O:-KHJ0l_C9G27_MkBwhfi:-1266247418) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/README.md#L272)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> we need to note that many apps may require separate server initialization files.
> 
> I don't appreciate having this stuff I wrote totally being deleted:
> 
> ```
>   In general, you may want different initialization for your server rendered components.
>  
> See below section on how to setup redux stores that allow multiple components to talk to the same store.
> ```</details>
I put a new version up. Let me know what you think.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[README.md, line 266 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIwA0jA11Fho2a3C1G:-KHJ0t1fJ8Ssn2PuwEHm:1304890241) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/README.md#L266)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> grammar</details>
fixed. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 2 of 5 files at r7.
Review status: 59 of 61 files reviewed at latest revision, 17 unresolved discussions.

---

*[CHANGELOG.md, line 14 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIujp_WSuKp5_0At6a:-KHJ1gXxcyxyt6C2LYPC:1066939641) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/CHANGELOG.md#L14)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> `compilition` typo</details>
@jbhatab still there.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/base_generator.rb, line 97 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIx8tiu0MjGO0ZxPw7:-KHJ10u9GCTJq4EfMJPj:893654194) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/base_generator.rb#L97)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> I have no idea what either of these changes these are</details>
I added it because they wouldn't be accessible unless we added another option. We could just split that off into a how-to. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 1 of 5 files at r7.
Review status: 59 of 61 files reviewed at latest revision, 12 unresolved discussions.

---

*[lib/generators/react_on_rails/base_generator.rb, line 97 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIx8tiu0MjGO0ZxPw7:-KHJ20Uj4Hbag0E0zUMu:-1053821800) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/base_generator.rb#L97)):*
<details><summary><i>Previously, jbhatab (Blaine Hatab) wrote…</i></summary>
> I added it because they wouldn't be accessible unless we added another option. We could just split that off into a how-to. </details>
how-to is appropriate.

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KHJ2PD816yzsJSFWySx:-1609311301) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
<details><summary><i>Previously, jbhatab (Blaine Hatab) wrote…</i></summary>
> @justin808 I want to add this in a separate PR or in the component generator PR. This PR is already getting a little big and we should merge this in first. I agree that it's a cool feature to show off. </details>
we still need to show passing _railsContext

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 1 of 5 files at r7.
Review status: 59 of 61 files reviewed at latest revision, 11 unresolved discussions.

---

*[lib/generators/react_on_rails/templates/base/base/client/.DS_Store, line  \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvRKdZoXfnl5CesMq:-KHJ2dGRroz5Z1eRAG7D:1550062816) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/.DS_Store#L)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> it _should_ be, but as evidenced by the fact that we are having this conversation, it obviously isn't, and therefore it would be prudent to prevent this from ever happening again by adding it to the gitignore. Considering the cost is nothing, and there is a benefit, it's a no-brainer. But this is sort of tangential to the point of the comment, which was "let's not commit a DS_Store file"</details>
why is this file still not removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt, line 3 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIvLsaZalOpWA2Jf_g:-KHJ2pad_qWm8uPIbJnE:-1627150518) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/Procfile.dev.tt#L3)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> still there on reviewable</details>
Weird. I have an extra line on mine and it's committed. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Still a few things...I can make the changes, or @jbhatab 

<details><summary><i>Previously, coveralls wrote…</i></summary>
> 
> [![Coverage Status](https://coveralls.io/builds/6096775/badge)](https://coveralls.io/builds/6096775)
> 
> Coverage increased (+0.09%) to 82.568% when pulling **2b9d125d3df29326f1ad6f3e581edb5f35b1d685 on feature/merge-server-client-webpack** into **949117978e75babd50f5f5d147f49c50674cf9cd on master**.
> </details>


---

Reviewed 1 of 5 files at r7.
Review status: 60 of 61 files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-:-KHJ2sx3Tv6aHbYG1th-:110622593)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 12 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIxU7PcwGluqhU6AYx:-KHJ3XEHZ5JQBgo02ru5:-1962625081) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L12)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> In that case I vote for the longer words so it will be just like the Rails envs</details>
changed to development.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

jbhatab commented:
*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KHJ49GVUP2sy6kzANss:1295012733) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> we still need to show passing _railsContext</details>
It failed CI for some reason. I can add an eslint ignore to this file but that feels off. We could disable it globally. Does it have file targeting in the config?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented on lib/generators/USAGE:
```diff
@@ -1,99 +1,13 @@
     to integrate the Redux state container framework. The necessary node modules
     will be automatically included for you.
 
-    The generator uses the organizational `paradigm of "bundles"`. These are like
```
Which file did this go to? or just removed, @jbhatab

justin808 commented on docs/api/ruby-api-hot-reload-view-helpers.md:
```diff
@@ -13,8 +13,7 @@ static vs. hot is picked based on whether `ENV["REACT_ON_RAILS_ENV"] == "HOT"`
  <!-- This is to load the hot assets. -->
- <%= env_javascript_include_tag(hot: ['http://localhost:3500/vendor-bundle.js',
-                                      'http://localhost:3500/app-bundle.js']) %>
+ <%= env_javascript_include_tag(hot: ['http://localhost:3500/webpack-bundle.js') %>
```
@jbhatab no reason to change this doc file. the point was to show that it can take multiple files. It doesn't need to correspond to the generated file, and you missed the closing square bracket.

justin808 commented on lib/generators/USAGE:
```diff
@@ -1,99 +1,13 @@
-setup involved when integrating Webpack with Rails. Defaults for options
-are such that the default is for the flag to be off. For example, the default for
-`-R` is that redux is off, and the default of -b is that skip-bootstrap is off.
+The react_on_rails:install generator integrates webpack with rails with ease. You can pass the redux option if you'd like to have redux setup for you automatically.
```
this is not a markdown file 

needed a CR

justin808 commented on lib/generators/react_on_rails/base_generator.rb:
```diff
@@ -104,39 +81,28 @@ def template_base_files
 # you could also consider creating symlinks there that point to the original
 # rails directories. In that case, you would not add these paths here.
-Rails.application.config.assets.precompile += %w( server-bundle.js )
+Rails.application.config.assets.precompile += %w( webpack-bundle.js )
```
This is not necessary if application.js has this file.

justin808 commented on spec/react_on_rails/test_helper/ensure_assets_compiled_spec.rb:
```diff
@@ -11,88 +11,25 @@
-        end
+      it "compiles the webpack assets" do
+        expect(compiler).to receive(:compile_assets).once
+        invoke_ensurer_with_doubles
```
where is this defined?

justin808 commented:
@jbhatab @robwise Please review my last commit. I'm going to create a beta release and test on the tutorial app.

justin808 commented:
@samnang, @jbhatab @robwise can you review https://github.com/shakacode/react_on_rails/pull/398/commits/a9f1f388e0f40fa897653618eec86e1894fc6c43

This is deployed live at http://www.reactrails.com.

Using: https://github.com/shakacode/react-webpack-rails-tutorial/pull/287

jbhatab commented:
*[lib/generators/USAGE, line 3 \[r10\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5NpMlf1zUEPR5TvIK-r10-3:-KHQkYggBKvm8iLopDrQ:160372363) ([raw file](https://github.com/shakacode/react_on_rails/blob/a26f399383a98ec7efe9df9ea086527061d1db35/lib/generators/USAGE#L3)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> this is not a markdown file 
> 
> 
> 
> needed a CR</details>
CR?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Reviewed 2 of 3 files at r9, 10 of 15 files at r11, 6 of 10 files at r12, 4 of 4 files at r13.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[lib/generators/USAGE, line 3 \[r10\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5NpMlf1zUEPR5TvIK-r10-3:-KHQm-W8QqgWv1Ogjy7J:24827671) ([raw file](https://github.com/shakacode/react_on_rails/blob/a26f399383a98ec7efe9df9ea086527061d1db35/lib/generators/USAGE#L3)):*
<details><summary><i>Previously, jbhatab (Blaine Hatab) wrote…</i></summary>
> CR?</details>
carriage return

---

*[lib/react_on_rails/test_helper/ensure_assets_compiled.rb, line 48 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQmWAImeyUHRd4fWfK:-KHQmWAImeyUHRd4fWfL:189061543) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/lib/react_on_rails/test_helper/ensure_assets_compiled.rb#L48)):*
> ```Ruby
> 
> React on Rails will ensure your JavaScript generated files are up to date, using your
> /client level package.json `build:test` command.
> ```

what's with the slash here, we want the line return preserved since this is a console message

---

*[lib/tasks/assets.rake, line 113 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQmkreNUIgqzOg6GT9:-KHQmkrfk5rmfiEsFBMk:183642096) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/lib/tasks/assets.rake#L113)):*
> ```Ruby
> # puts "Enhancing assets:precompile with react_on_rails:assets:compile_environment"
> # Rake::Task["assets:precompile"]
> #   .clear_prerequisites
> ```

@justin808 did we ever figure out what `clear_prerequisites` does?

---

*[spec/dummy/config/initializers/react_on_rails.rb, line 23 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQn6r4qIxMCiDOu1yy:-KHQn6r4qIxMCiDOu1yz:-1398763344) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/spec/dummy/config/initializers/react_on_rails.rb#L23)):*
> ```Ruby
>   # If you are using the same file for client and server rendering, having this set probably does
>   # not affect performance.
>   config.server_bundle_js_file = "server-bundle.js"
> ```

shouldn't this be set to `webpack-bundle.js` by default, since by default we don't generate separate bundles?

---

*[spec/react_on_rails/test_helper/ensure_assets_compiled_spec.rb, line 16 \[r10\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG_K1hJSTy1WijeX_a7-r10-16:-KHQnXBNlb9pf0h_MhER:-322193055) ([raw file](https://github.com/shakacode/react_on_rails/blob/a26f399383a98ec7efe9df9ea086527061d1db35/spec/react_on_rails/test_helper/ensure_assets_compiled_spec.rb#L16)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> where is this defined?</details>
![where it's defined](https://s3.amazonaws.com/f.cl.ly/items/0r1I2h090B252q3z1U2c/Image%202016-05-10%20at%201.53.45%20PM.png?v=b617197c)

---

*[lib/generators/react_on_rails/templates/base/base/client/.DS_Store, line 0 \[r7\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQnmndWT6Q9GB83D1I:-KHQnmndWT6Q9GB83D1J:2008681784) ([raw file](https://github.com/shakacode/react_on_rails/blob/bbc35752787dc8d7bbb908260af4df9e9c2e9d87/lib/generators/react_on_rails/templates/base/base/client/.DS_Store#L0)):*
this is still in the commit, pls delete

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Reviewed 1 of 7 files at r4, 5 of 8 files at r5, 5 of 16 files at r6, 2 of 5 files at r7.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[lib/generators/react_on_rails/templates/base/base/.DS_Store, line 0 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQoajKuqqRIeh5z-Ou:-KHQoajLrt-bKrgdmk6H:-715374630) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/lib/generators/react_on_rails/templates/base/base/.DS_Store#L0)):*
.DS_Store

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

robwise commented:
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[lib/generators/react_on_rails/base_generator.rb, line 97 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIx8tiu0MjGO0ZxPw7:-KHQrMbu05-npTlETIR9:1843914470) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/base_generator.rb#L97)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> how-to is appropriate.</details>
let's do that, I think these files will definitely confuse newbies

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Ready? I think so.

<details><summary><i>Previously, coveralls wrote…</i></summary>
> 
> [![Coverage Status](https://coveralls.io/builds/6117503/badge)](https://coveralls.io/builds/6117503)
> 
> Coverage increased (+0.07%) to 82.544% when pulling **f80eb14cc86cab51689df040ea7a63f484054007 on feature/merge-server-client-webpack** into **56c0c8d8b56df4b9d537b8c5fbe86274efba50fe on master**.
> </details>


---

Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[lib/generators/react_on_rails/base_generator.rb, line 97 \[r6\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHIx8tiu0MjGO0ZxPw7:-KHJZ9yhPa0WwD-8gEwi:-1519380406) ([raw file](https://github.com/shakacode/react_on_rails/blob/2b9d125d3df29326f1ad6f3e581edb5f35b1d685/lib/generators/react_on_rails/base_generator.rb#L97)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary>
> how-to is appropriate.</details>
we're not going copy either.

---

*[lib/generators/react_on_rails/templates/base/base/.DS_Store, line  \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQoajKuqqRIeh5z-Ou:-KHTqYGnJC4olQWSXSSw:-1054325829) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/lib/generators/react_on_rails/templates/base/base/.DS_Store#L)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> .DS_Store</details>
removed

---

*[lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt, line 12 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KG5W88mqgpk5LwIkmon:-KHTrA1qpog8rpYeJ6hm:-497103488) ([raw file](https://github.com/shakacode/react_on_rails/blob/8401c9d85f2c2fe8ff3d6153680afe4086198c13/lib/generators/react_on_rails/templates/redux/base/client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx.tt#L12)):*
<details><summary><i>Previously, jbhatab (Blaine Hatab) wrote…</i></summary>
> It failed CI for some reason. I can add an eslint ignore to this file but that feels off. We could disable it globally. Does it have file targeting in the config?</details>
Probably had to use a underscore. I changed this and added commentsjj

---

*[lib/react_on_rails/test_helper/ensure_assets_compiled.rb, line 48 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQmWAImeyUHRd4fWfK:-KHTrVZxaaYYwuW9t-an:-1276463811) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/lib/react_on_rails/test_helper/ensure_assets_compiled.rb#L48)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> what's with the slash here, we want the line return preserved since this is a console message</details>
to mean this is the top level client directory

---

*[lib/tasks/assets.rake, line 113 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQmkreNUIgqzOg6GT9:-KHTrbrUGuNtHi6vJQaF:-1427549483) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/lib/tasks/assets.rake#L113)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> @justin808 did we ever figure out what `clear_prerequisites` does?</details>
yes -- removes the stuff to do before the task 

I documented in this file on why it's needed above.

---

*[spec/dummy/config/initializers/react_on_rails.rb, line 23 \[r13\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-KHQn6r4qIxMCiDOu1yy:-KHTrjdI6FsCGG_0dlOs:1310469762) ([raw file](https://github.com/shakacode/react_on_rails/blob/f80eb14cc86cab51689df040ea7a63f484054007/spec/dummy/config/initializers/react_on_rails.rb#L23)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary>
> shouldn't this be set to `webpack-bundle.js` by default, since by default we don't generate separate bundles?</details>
the spec dummy example uses separate files

we didn't change that (no diff mark)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/398#-:-KHTrwXNKUXuS6Br-iKe:-1722611506)*
<!-- Sent from Reviewable.io -->
